alf/networks/actor_networks.py

```diff
@@ -136,10 +138,34 @@ def forward(self, observation, state=()):
         encoded_obs, _ = self._encoding_net(observation)
 
         actions = []
+        i = 0
         for layer, spec in zip(self._action_layers, self._flat_action_spec):
-            action = layer(encoded_obs)
+            _info = {} if alf.summary.should_summarize_output() else None
+            action = layer(encoded_obs, info=_info)
+            if alf.summary.should_summarize_output():
+                name = ('summarize_output/' + self.name + '.action_layer.' +
+                        str(i) + '.pre_activation.output_norm')
+                if not self.training:
```

Comment:
Where is this self.training defined?

Reply:
This is from nn.Module, set by nn.Module.train(mode).